libraries/type-length-value/src/discriminator.rs

```diff
@@ -0,0 +1,54 @@
+//! Discriminator for differentiating account types, the "Type" in the
+//! Type-Length-Value structure. Since the word "type" is reserved in Rust,
+//! we use the term "Discriminator" and "Type" interchangeably.
+
+use {
+    bytemuck::{Pod, Zeroable},
+    solana_program::program_error::ProgramError,
+};
+
+/// Trait to be implemented by all value types in the TLV structure, specifying
+/// just the discriminator
+pub trait TlvType {
+    /// Associated value type enum, checked at the start of TLV entries
+    const TYPE: Discriminator;
```

Comment:
As mentioned, this isn't possible unfortunately! And the base is always the same, since it's just discriminator + length

Reply:
Yeah sure, I'll go with `DISCRIMINATOR`, since eventually doing something like `<MyType as TlvType>::TLV_DISCRIMINATOR` seems a bit verbose

Reply:
I think it's probably better to prefix with `TLV`, since anchor already exports a trait called `Discriminator` that requires a DISCRIMINATOR type.

https://github.com/coral-xyz/anchor/blob/c14ae6b4f8ac284c155cb146436c6de6e59c4603/lang/src/lib.rs#L205

Reply:
As discussed offline, I now agree with you, thanks for the suggestion!